src/USER-GWBC/random_extra.cpp

```diff
@@ -0,0 +1,130 @@
+/* ----------------------------------------------------------------------

```

Comment:
why copy the pRNG class and not either create a derived class with only the extra functions added, or simply add the additional functionality to the original pRNG?